src/shared/components/data-table/components/main/components/table-row/TableRow.test.tsx

```diff
@@ -0,0 +1,137 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import {
+  TableContext,
+  type TableContextType
+} from 'src/shared/components/data-table/DataTable';
+
+import TableRow from './TableRow';
+import { defaultDataTableState } from 'src/shared/components/data-table/dataTableReducer';
+import { createDataTableSampleData } from 'src/shared/components/data-table/DataTable.test';
+
+const { columns, data } = createDataTableSampleData(1, 10);
+
+const defaultProps = {
+  ...defaultDataTableState,
+  data,
+  dispatch: jest.fn(),
+  columns,
+  rows: data.map((row, index) => ({
+    rowId: index,
+    cells: row
+  }))
+};
+
+describe('<TableRow />', () => {
+  const renderTableRow = (props: Partial<TableContextType> = {}) =>
+    render(
+      <TableContext.Provider value={{ ...defaultProps, ...props }}>
+        <table>
+          <tbody>
+            <TableRow
+              rowId={String(defaultProps.rows[0].rowId)}
+              rowData={defaultProps.rows[0].cells}
+            />
+          </tbody>
+        </table>
+      </TableContext.Provider>
+    );
+
+  let container: HTMLElement;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('renders without error', () => {
+    container = renderTableRow().container;
+    expect(() => container).not.toThrow();
+  });
+
+  it('respects hiddenRowIds and renders nothing', () => {
+    container = renderTableRow({
+      hiddenRowIds: { [defaultProps.rows[0].rowId]: true }
```

Comment:
Didn't have this question earlier; but looking at the test now — why is this shaped as a `Record<string, boolean>`? The `false` value wouldn't make sense here, would it? This looks something like a list of ids. In the old days, we would have used an array for this; but these days, a Set would be a better option.

Reply:
I agree that Set would fit better here. I'll update it